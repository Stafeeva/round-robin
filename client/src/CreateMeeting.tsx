import React, { FC } from "react";
import { Button, Input } from "antd";
import { CopyOutlined } from "@ant-design/icons";
import { Flex } from "antd";

// "/create"

const CreateMeeting: FC = () => {
  return (
    <Flex
      style={{ width: "100%", height: "100vh" }}
      align="center"
      vertical
      justify="space-around"
    >
      <Flex
        vertical
        gap="middle"
        style={{
          width: "400px",
          border: "1px solid #1677ff",
          borderRadius: "8px",
          padding: "72px",
        }}
      >
        <Flex justify="space-between" align="center">
          Meeting name
          <Input value="Apollo team standup" style={{ width: "70%" }} />
        </Flex>
        <Flex justify="space-between" align="center" gap="6px">
          Meeting code
          <Input value="xyz-d89-s123" style={{ width: "60%" }} />
          <Button type="default" icon={<CopyOutlined />} />
        </Flex>
        <Flex justify="space-between" align="center" gap="6px">
          Speaker duration
          <Flex justify="space-around" gap="6px">
            <Button type="default">1m</Button>
            <Button type="primary">2m</Button>
            <Button type="default">3m</Button>
            <Button type="default">5m</Button>
          </Flex>
        </Flex>
        <Flex justify="space-between" align="center" gap="6px">
          Auto proceed to next speaker
          <Flex justify="space-around" gap="6px">
            <Button type="primary">Yes</Button>
            <Button type="default">No</Button>
          </Flex>
        </Flex>
        <Button type="primary" style={{ marginTop: "48px" }}>
          Create
        </Button>
      </Flex>
    </Flex>
  );
};

export default CreateMeeting;
