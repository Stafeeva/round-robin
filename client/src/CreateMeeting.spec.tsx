import React from "react";
import { render } from "@testing-library/react";

import CreateMeeting from "./CreateMeeting";

const createMeeting = () => render(<CreateMeeting />);

// test Create meeting renders
test("Create meeting component renders", () => {
  const { container } = createMeeting();
  expect(container).toMatchSnapshot();
});

test("Allows user to input meeting name", () => {
  const { getByText } = createMeeting();
  const input = getByText("Meeting name");
  expect(input).toBeInTheDocument();
});

test("Allows user to copy meeting code", () => {
  const { getByText } = createMeeting();
  const input = getByText("Meeting code");
  expect(input).toBeInTheDocument();
});

test("User can select a meeting duration", () => {
  const { getByText } = createMeeting();
  const input = getByText("Speaker duration");
  expect(input).toBeInTheDocument();
});

test("User can select to auto proceed to next speaker", () => {
  const { getByText } = createMeeting();
  const input = getByText("Auto proceed to next speaker");
  expect(input).toBeInTheDocument();
});

test("User can create a meeting", () => {
  const { getByText } = createMeeting();
  const input = getByText("Create");
  expect(input).toBeInTheDocument();
});
