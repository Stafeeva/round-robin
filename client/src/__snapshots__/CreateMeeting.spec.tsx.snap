// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create meeting component renders 1`] = `
<div>
  <div
    class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center ant-flex-justify-space-around ant-flex-vertical"
    style="width: 100%; height: 100vh;"
  >
    <div
      class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-stretch ant-flex-gap-middle ant-flex-vertical"
      style="width: 400px; border: 1px solid #1677ff; border-radius: 8px; padding: 72px;"
    >
      <div
        class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center ant-flex-justify-space-between"
      >
        Meeting name
        <input
          class="ant-input css-dev-only-do-not-override-1uq9j6g ant-input-outlined"
          style="width: 70%;"
          type="text"
          value="Apollo team standup"
        />
      </div>
      <div
        class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center ant-flex-justify-space-between"
        style="gap: 6px;"
      >
        Meeting code
        <input
          class="ant-input css-dev-only-do-not-override-1uq9j6g ant-input-outlined"
          style="width: 60%;"
          type="text"
          value="xyz-d89-s123"
        />
        <button
          class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default ant-btn-icon-only"
          type="button"
        >
          <span
            class="ant-btn-icon"
          >
            <span
              aria-label="copy"
              class="anticon anticon-copy"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="copy"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M832 64H296c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h496v688c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8V96c0-17.7-14.3-32-32-32zM704 192H192c-17.7 0-32 14.3-32 32v530.7c0 8.5 3.4 16.6 9.4 22.6l173.3 173.3c2.2 2.2 4.7 4 7.4 5.5v1.9h4.2c3.5 1.3 7.2 2 11 2H704c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32zM350 856.2L263.9 770H350v86.2zM664 888H414V746c0-22.1-17.9-40-40-40H232V264h432v624z"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
      <div
        class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center ant-flex-justify-space-between"
        style="gap: 6px;"
      >
        Speaker duration
        <div
          class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-justify-space-around"
          style="gap: 6px;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default"
            type="button"
          >
            <span>
              1m
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-primary"
            type="button"
          >
            <span>
              2m
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default"
            type="button"
          >
            <span>
              3m
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default"
            type="button"
          >
            <span>
              5m
            </span>
          </button>
        </div>
      </div>
      <div
        class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-align-center ant-flex-justify-space-between"
        style="gap: 6px;"
      >
        Auto proceed to next speaker
        <div
          class="ant-flex css-dev-only-do-not-override-1uq9j6g ant-flex-justify-space-around"
          style="gap: 6px;"
        >
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-primary"
            type="button"
          >
            <span>
              Yes
            </span>
          </button>
          <button
            class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-default"
            type="button"
          >
            <span>
              No
            </span>
          </button>
        </div>
      </div>
      <button
        class="ant-btn css-dev-only-do-not-override-1uq9j6g ant-btn-primary"
        style="margin-top: 48px;"
        type="button"
      >
        <span>
          Create
        </span>
      </button>
    </div>
  </div>
</div>
`;
